test/finch_test.exs

```diff
@@ -392,9 +398,9 @@ defmodule FinchTest do
 
       try do
         Finch.build(:get, endpoint(bypass)) |> Finch.request(client, pool_timeout: 0)
-      catch
-        :exit, reason ->
-          assert {:timeout, _} = reason
+      rescue
+        e in RuntimeError ->
+          assert e.message =~ "Finch was unable to provide a connection within the timeout"
```

Comment:
you may consider converting it to a more common:

```elixir
assert_raise RuntimeError, ~r/Finch was unable to provide a connection within the timeout/ fn ->
  # ...
end
```

That being said, and this is out of the scope of this PR, but curious if instead of raising an error have you considered returning an `{:error, %RuntimeError{}}` tuple instead?

Reply:
I like the idea of changing this to an error tuple instead of raising. I think this only raises because NimblePool does, so we kept it the same here

Reply:
I would prefer to raise instead of a tuple because I can't think of a scenario where you want to handle this. Even if you say, "well, I want to retry", that's bound to make things worse. :)

Reply:
That definitely makes sense

Reply:
fixed!